core/src/main/java/org/apache/spark/util/collection/unsafe/sort/UnsafeExternalSorter.java

```diff
@@ -521,4 +522,60 @@ public long getKeyPrefix() {
       return upstream.getKeyPrefix();
     }
   }
+
+  /**
+   * Returns a iterator. It is the caller's responsibility to call `cleanupResources()`
```

Comment:
Can you document what the difference is between this iterator and the sorted iterator? Is it simply that one is sorted and the other is not?